FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -1388,87 +1925,175 @@ BaseType_t xReturn = pdTRUE;
 				}
 			}
 
-#if( ipconfigUSE_LLMNR == 1 )
-			else if( ( usQuestions != ( uint16_t ) 0U ) && ( usType == dnsTYPE_A_HOST ) && ( usClass == dnsCLASS_IN ) && ( pcRequestedName != NULL ) )
+	#if( ipconfigUSE_LLMNR == 1 )
+				else if( ( usQuestions != ( uint16_t ) 0U ) && 
+						 ( ( usType == dnsTYPE_A_HOST ) || ( usType == dnsTYPE_AAAA_HOST ) ) &&
+						 ( usClass == dnsCLASS_IN ) &&
+						 ( pcRequestedName != NULL ) )
 			{
 				/* If this is not a reply to our DNS request, it might an LLMNR
 				request. */
-				if( xApplicationDNSQueryHook( &( pcRequestedName[ 1 ] ) )  != pdFALSE )
+			NetworkBufferDescriptor_t *pxNetworkBuffer;
+			NetworkEndPoint_t *pxEndPoint, xEndPoint;
+			int16_t usLength;
+			LLMNRAnswer_t *pxAnswer;
+			size_t uxDataLength;
+			size_t uxExtraLength;
+			size_t uxOffsets[ 3 ];
+			uint8_t *pucNewBuffer;
```

Comment:
Defensive programming opportunity. Can we initialize `pucNewBuffer` to `NULL`?

Reply:
It will be set unconditionally before it is read. But I initialised it, assuming that the compiler will have its own thoughts about it.